nano/node/websocket.cpp

```diff
@@ -4,6 +4,45 @@
 #include <nano/node/node.hpp>
 #include <nano/node/websocket.hpp>
 
+nano::websocket::confirmation_options::confirmation_options (boost::property_tree::ptree const & options_a) :
+all_local_accounts (options_a.get<bool> ("all_local_accounts", false))
+{
+	auto accounts_l (options_a.get_child_optional ("accounts"));
+	if (accounts_l)
+	{
+		for (auto account : *accounts_l)
+		{
+			// Check if the account is valid, but no error handling if it's not, simply not added to the filter
+			nano::account result (0);
+			if (!result.decode_account (account.second.data ()))
+			{
+				accounts.insert (account.second.data ());
+			}
+		}
+	}
+}
+
+bool nano::websocket::confirmation_options::filter (boost::property_tree::ptree const & message_a, nano::node & node_a) const
+{
+	// If this fails, the message builder has been changed
+	auto account_text (message_a.get<std::string> ("message.account"));
+	if (all_local_accounts)
+	{
+		auto transaction (node_a.wallets.tx_begin_read ());
+		nano::account account (0);
+		assert (account.decode_account (account_text));
+		if (node_a.wallets.exists (transaction, account))
```

Comment:
I think this is a good idea, with the other use case being once you have a pruned node with only the local accounts, everything is kept in memory, blazing fast. 👍

Reply:
@clemahieu any thoughts on this? If it's hotspot data, lmdb is likely to bring these to the page cache, but there's still some overhead. Should probably profile this to avoid unnecessary complexity of maintaining a cache.

Reply:
Did some ad-hoc profiling in the `if (all_local_accounts)` scope, with `nano::autotimer` and instructions by @cryptocode . I have an NVME SSD.

- 1 wallet with 100k accounts: 14-16 µs, no observed trend for subsequent tries
- 20 wallets with 5000 accounts each: 38-50 µs, no observed trend
- 2000 wallets with 20 accounts each: 1100-1500 µs, no observed trend

All profiling done in a Debug build.

Reply:
Release build:

- 20 wallets with 5000 accounts each: 1 µs
- 3000 wallets with 20 accounts each: 200-400 µs, no observed trend

Reply:
Great data points! A huge amount of wallets should be rare (are there any examples?) and we can document that local-account filtering may affect performance if a large number of wallets are used. These numbers look acceptable to me, the limiting factor will be elsewhere.